### ISSUE TITLE
Pin map: no vector tile points with filters

### ISSUE THREAD

**Issue by danpelota** [metabase/metabase#9402]
**Browser**: Firefox 64.0, Chrome 71.0.3578.98, Safari 12.0.2 (all three)
**Local OS**: macOS 10.12.6
**Metabase OS**: Ubuntu 16.04.4 LTS
**Database** (both Metabase internal and datastore): PostgreSQL 10.4
**Metabase version**: v0.31.2
**Metabase deployment**: Running .jar file from ubuntu command line

Steps to repeat the issue:

1) View "raw data" on a table with 133k rows and columns for latitude and longitude
2) Select Visualization: Map. Points render correctly:
![screen shot 2019-02-11 at 11 56 40 am](https://user-images.githubusercontent.com/724968/52579472-3c647f00-2df4-11e9-9d5e-d1a7c90619fb.png)
3) Add any filter to the map that results in more than a few thousand records, either through the filter widget or drawing a box, and the rendered points vanish:
![screen shot 2019-02-11 at 11 58 51 am](https://user-images.githubusercontent.com/724968/52579573-6b7af080-2df4-11e9-9432-617406e93526.png)

Note that the map renders correctly if the filter results in a count of rows small enough to visualize as map markers. Here's the same query, with an additional selective filter that results in 488 rows:

![screen shot 2019-02-11 at 12 00 35 pm](https://user-images.githubusercontent.com/724968/52579693-a9781480-2df4-11e9-8d20-51a8500c72a4.png)

Looking at the network requests, it seems like the issue is in the call to the map tile API. Here's one of the requests from the query that results in no visible tiles:

Request: 
`/api/tiles/5/9/12/1231/1208/13/15/?query=%7B%22query%22%3A%7B%22source-table%22%3A56%2C%22filter%22%3A%5B%22inside%22%2C1231%2C1208%2C42.318020788513444%2C-121.93005093153761%2C23.02105533364351%2C-79.54533140291808%5D%7D%2C%22type%22%3A%22query%22%2C%22database%22%3A2%7D`

Response:
```
{  
   "value":[  
      "and",
      [  
         "inside",
         1231,
         1208,
         42.318020788513444,
         -121.93005093153761,
         23.02105533364351,
         -79.54533140291808
      ],
      [  
         "inside",
         [  
            "field-id",
            1231
         ],
         [  
            "field-id",
            1208
         ],
         40.97989944013221,
         -78.74999809311707,
         31.952163393282277,
         -67.49999836552892
      ]
   ],
   "error":"schema.utils.NamedError@7e763b06",
   "message":"Output of simplify-compound-filter does not match schema: \n\n\t \u001B[0;33m  (named [nil (named (named (not (some-matching-condition? a-clojure.lang.PersistentVector)) \"Must be a valid instance of one of these clauses: :and, :or, :not, :=, :!=, :<, :>, :<=, :>=, :between, :starts-with, :ends-with, :contains, :does-not-contain, :inside, :is-null, :not-null, :time-interval, :segment\") \"first-clause\") nil] \"Must be a valid instance of one of these clauses: :and, :or, :not, :=, :!=, :<, :>, :<=, :>=, :between, :starts-with, :ends-with, :contains, :does-not-contain, :inside, :is-null, :not-null, :time-interval, :segment\") \u001B[0m \n\n",
   "type":"class clojure.lang.ExceptionInfo",
   "stacktrace":[  
      "--> mbql.util$fn__18799$simplify_compound_filter__18804.invoke(util.clj:182)",
      "mbql.util$fn__18902$combine_filter_clauses__18907$fn__18908.invoke(util.clj:234)",
      "mbql.util$fn__18902$combine_filter_clauses__18907.doInvoke(util.clj:230)",
      "api.tiles$query_with_inside_filter.invokeStatic(tiles.clj:62)",
      "api.tiles$query_with_inside_filter.invoke(tiles.clj:50)",
      "util$rpartial$fn__4216.doInvoke(util.clj:185)",
      "api.tiles$fn__55758.invokeStatic(tiles.clj:137)",
      "api.tiles$fn__55758.invoke(tiles.clj:117)",
      "middleware$enforce_authentication$fn__56174.invoke(middleware.clj:113)",
      "api.routes$fn__56370.invokeStatic(routes.clj:87)",
      "api.routes$fn__56370.invoke(routes.clj:87)",
      "routes$fn__56409$fn__56410.doInvoke(routes.clj:108)",
      "routes$fn__56409.invokeStatic(routes.clj:103)",
      "routes$fn__56409.invoke(routes.clj:103)",
      "middleware$catch_api_exceptions$fn__56309.invoke(middleware.clj:436)",
      "middleware$log_api_call$fn__56287$fn__56289.invoke(middleware.clj:364)",
      "middleware$log_api_call$fn__56287.invoke(middleware.clj:363)",
      "middleware$add_security_headers$fn__56229.invoke(middleware.clj:252)",
      "core$wrap_streamed_json_response$fn__62673.invoke(core.clj:67)",
      "middleware$bind_current_user$fn__56179.invoke(middleware.clj:137)",
      "middleware$maybe_set_site_url$fn__56239.invoke(middleware.clj:290)",
      "middleware$add_content_type$fn__56232.invoke(middleware.clj:262)"
   ]
}
```

**Comment by flamber** [comment]
This is a duplicate of #4394, but possibly better described in #7560

**Comment by danpelota** [comment]
Good point. I noticed that while the symptom is identical, it seems like the underlying cause may be different:

- In #4394, the issue seems to be a 404 returned by the API call. The request URL is a bit different (contains `.../undefined/undefined/...`)
- In #7560, the response indicates "API endpoint does not exist" and the issue is only reproduced on SQL/native queries, not the query builder.

In my issue, the 500 response contains the message:

> Output of simplify-compound-filter does not match schema: ... "Must be a valid instance of one of these clauses: :and, :or, :not, :=, :!=, :<, :>, :<=, :>=, :between, :starts-with, :ends-with, :contains, :does-not-contain, :inside, :is-null, :not-null, :time-interval, :segment"...

With that said, I'm happy to move the discussion to one of those issues if they're similar enough to be merged.

**Comment by flamber** [comment]
You probably have the best described issue of all three. And yes, you're seeing some different errors, but I've just tried to reproduce your issue and then found the two others, which I could also reproduced.
Given that it fails for the same reason - above 999 point and using a filter (`WHERE`) - leads me to think it's bound in the same problem.
Keep your issue open until one of the Metabase team members comes by, since I really like the way you described the issue.

**Comment by tlrobinson** [comment]
This will be fixed by #9419.

#4394 and #7560 look like they are both for native queries so I'll leave one of those open.
